src/gui/setuserstatusdialogmodel.cpp

```diff
@@ -0,0 +1,433 @@
+/*
+ * Copyright (C) by Felix Weilbach <felix.weilbach@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include "setuserstatusdialogmodel.h"
+
+#include <ocsuserstatusconnector.h>
+#include <userstatusconnector.h>
+#include <theme.h>
+
+#include <QDateTime>
+#include <QLoggingCategory>
+
+#include <algorithm>
+#include <cmath>
+#include <cstddef>
+
+namespace OCC {
+
+Q_LOGGING_CATEGORY(lcUserStatusDialogModel, "nextcloud.gui.userstatusdialogmodel", QtInfoMsg)
+
+SetUserStatusDialogModel::SetUserStatusDialogModel(QObject *parent)
+    : QObject(parent)
+    , _dateTimeProvider(new DateTimeProvider)
```

Comment:
use `std::make_unique<DateTimeProvider>()`

Reply:
What is the benefit?

Reply:
the object dynamically allocated by `std::make_unique<DateTimeProvider>()` is already managed by an instance of `std::unique_ptr` so it can never leak
direct usage of `new` can lead to a leak if there is something happening like an exception being fired between the call to new and the call to the constructor of the `std::unique_ptr`

Reply:
Not sure because that is the implementation of `std::make_unique()` in my stl:

```cpp
  /// std::make_unique for single objects
  template<typename _Tp, typename... _Args>
    inline typename _MakeUniq<_Tp>::__single_object
    make_unique(_Args&&... __args)
    { return unique_ptr<_Tp>(new _Tp(std::forward<_Args>(__args)...)); }
```

Seems to be just that what I do.